### PR TITLE
fix: js apps filter in enum usecase search method

### PR DESCRIPTION
**Pull request by utkarsh-dixit**
# 🔍 Review Summary 
 ### Purpose
- Address a bug in the JavaScript SDK related to the `advancedUseCaseSearch` method, improving the robustness of the search functionality.

### Key Changes
- **Bug Fix**: 
  - Enhanced the `advancedUseCaseSearch` method by adding a nullish coalescing operator to safely access the `apps` array, preventing runtime errors when `apps` is undefined.

### Impact
- Improves the reliability of the search functionality in the JavaScript SDK by preventing potential runtime errors, enhancing the overall user experience. 

<details><summary>Original Description</summary>

No existing description found

</details>

### CODE REVIEW

File: js/src/sdk/models/actions.ts [hunk 1]
```diff
@@ -276,7 +276,7 @@ export class Actions {
     try {
       const { data: res } = await apiClient.actionsV2.advancedUseCaseSearch({
         query: {
-          apps: data.apps.join(","),
+          apps: data.apps?.join(","),
```
Comment (shreysingla11) [inline js/src/sdk/models/actions.ts:279]: Good fix for handling undefined `data.apps`. Consider adding a default value to handle empty string case: `data.apps?.join(",") || ""`. This would prevent potential issues with the API expecting a string value.

**Comment by shreysingla11** [comment]
### Code Review Summary

#### Changes Overview
- Fixed potential undefined error in `findActionEnumsByUseCase` method by adding optional chaining for `data.apps.join()`
- The change improves error handling when `data.apps` is undefined

#### Code Quality Assessment
✅ **Bug Fix**: The change correctly addresses the potential runtime error
✅ **Minimal Impact**: The change is focused and doesn't introduce unnecessary modifications
✅ **Type Safety**: The fix is consistent with TypeScript's type safety features

#### Suggestions for Future Improvements
1. Consider adding a default empty string for the apps parameter
2. Add type definitions for the expected input parameters
3. Add test cases covering undefined/null scenarios

Overall, this is a good fix that improves the code's robustness. The changes are minimal and focused on solving a specific issue.

**Review by himanshu-dixit — APPROVED** [review]
